src/components/components/footer/footer.js

```diff
@@ -1,159 +1,171 @@
 import MediaQuery from 'react-responsive';
+import LayoutContained from '../../layouts/layout-contained/layout-contained';
+import Navigation from '../../navigation/footer-navigation/footer-navigation';
 
-// Local modules.
-import LayoutContained from './../../layouts/layout-contained/layout-contained';
-import Navigation from './../../navigation/footer-navigation/footer-navigation';
+class Footer extends React.PureComponent {
+  constructor(props) {
+    super(props);
+  }
 
-// Styles.
-import './footer.scss';
+  isUserLoggedIn = () => {
+    return !!this.props.user.user;
+  };
 
-const Footer = props => {
-  return (
-    <div className="footer">
-      <LayoutContained className="footer__inner">
-        <nav className="footer__primary">
-          <Link
-            to="/"
-            className="footer__primary-logo"
-            activeClassName="is-active"
-          >
-            <span className="footer__ossn-logo">
-              Open Source Student Network
-            </span>
-            <span className="footer__mozilla-logo">Mozilla</span>
-          </Link>
+  render() {
+    // trivially added link list.
+    const footerNavigationList1 = [
+      {
+        title: 'Clubs',
+        target: '/clubs'
+      },
+      {
+        title: 'Members',
+        target: '/members'
+      },
+      {
+        title: 'Opportunities',
+        target: '/contribute'
+      },
+      this.isUserLoggedIn()
+        ? {
+            title: 'Logout',
+            logout: true
+          }
+        : {
+            title: 'Log In/SignUp',
+            login: true
+          },
+      {
+        title: 'Affiliations',
+        target: '/organizations'
+      },
+      {
+        title: 'About Us',
+        target: '/about'
+      }
+    ];
 
-          <div className="footer__sections">
-            <MediaQuery minWidth={480} className="footer__section-item">
-              <Navigation
-                title="Navigation"
-                links={footerNavigationList1}
-                className="footer__section--navigation"
-                id="footer-section-navigation"
-              />
-              <Navigation
-                title="Keep in touch"
-                links={footerNavigationList2}
-                className="footer__section--social"
-                id="footer-section-social"
-              />
-            </MediaQuery>
-            <MediaQuery maxWidth={479} className="footer__section-item">
-              <Navigation
-                title="Navigation"
-                isMobile
-                links={footerNavigationList1}
-                className="footer__section--navigation"
-              />
-              <Navigation
-                title="Keep in touch"
-                isMobile
-                links={footerNavigationList2}
-                className="footer__section--social"
-              />
-            </MediaQuery>
-          </div>
-        </nav>
+    const footerNavigationList2 = [
```

Comment:
There is nothing dynamic in this list so it doesn't need to be inside the render method, it can be outside of the component